lib/fauna/query.rb

```diff
@@ -89,15 +89,15 @@ def self.quote(expr)
     #   If this takes more than one argument, the lambda destructures an array argument.
     #   (To destructure single-element arrays use ::lambda_expr.)
     def self.lambda(&block)
-      n_args = block.arity
-      fail ArgumentError, 'Function must take at least 1 argument.' if n_args == 0
-
-      with_auto_vars(n_args) do |vars|
-        if n_args == 1
-          lambda_expr vars[0], block.call(var(vars[0]))
-        else
-          lambda_expr vars, block.call(*(vars.map { |v| var(v) }))
-        end
+      vars = block_parameters block
+      n_args = vars.length
+      case n_args
+      when 0
+        fail ArgumentError, 'Block must take at least 1 argument.'
+      when 1
```

Comment:
We don't use an array pattern in the case of only one var. `:foo`, not `[:foo]`.